gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -0,0 +1,195 @@
+
+  def update(self, to_merge):
+    """Updates ``self``'s headers with values from ``to_merge``.
+
+    If a specific key does not already exists in a specific one of ``self``'s
+    headers, that key and the associated value will be added. If the key does
+    already exist in the specific header of ``self``, then the value of that key
+    will be updated with the value from ``to_merge``.
+
+    Args:
+      to_merge (:class:`VcfHeader`): The VcfHeader object that's headers will be
+        merged into the headers of self.
+    """
+    if not isinstance(to_merge, VcfHeader):
+      raise NotImplementedError
+
+    self._merge_header_fields(self.infos, to_merge.infos)
+    self._merge_header_fields(self.filters, to_merge.filters)
+    self._merge_header_fields(self.alts, to_merge.alts)
+    self._merge_header_fields(self.formats, to_merge.formats)
+    self._merge_header_fields(self.contigs, to_merge.contigs)
+
+  def __eq__(self, other):
+    return (self.infos == other.infos and
+            self.filters == other.filters and
+            self.alts == other.alts and
+            self.formats == other.formats and
+            self.contigs == other.contigs)
+
+  def __repr__(self):
+    return ', '.join([str(header) for header in [self.infos,
+                                                 self.filters,
+                                                 self.alts,
+                                                 self.formats,
+                                                 self.contigs]])
+
+  def _values_asdict(self, header):
+    """Converts pyVCF header values to dictionaries."""
+
+    # These methods were not designed to be protected. They start with an
+    # underscore to avoid confilcts with field names. For more info, see below:
+    # https://docs.python.org/2/library/collections.html#collections.namedtuple
+    return {key: header[key]._asdict() for key in header}  # pylint: disable=W0212
+
+  def _merge_header_fields(self, source, to_merge):
+    """Modifies ``source`` to add any keys from ``to_merge`` not in ``source``.
+
+    Args:
+      source (dict): Source header fields.
+      to_merge (dict): Header fields to merge with ``source``.
+    Raises:
+      ValueError: If the header fields are incompatible (e.g. same key with
+        different types or numbers).
+    """
+    for key, to_merge_value in to_merge.iteritems():
+      if key not in source:
+        source[key] = to_merge_value
+      else:
+        source_value = source[key]
+        # Verify num and type fields are the same for infos and formats
+        if (source_value.keys() != to_merge_value.keys()
+            or ('num' in source_value and 'type' in source_value
+                and (source_value['num'] != to_merge_value['num']
+                     or source_value['type'] != to_merge_value['type']))):
+          raise ValueError(
+              'Incompatible number of types in header fields: %s, %s' % (
```

Comment:
Please replace with: "Incompatible number or type in ..."  ... just realized I had the same error in my code :)

Reply:
Done.